### DISCUSSION TITLE
Does any other tool have the persistence / order feature?

### DISCUSSION THREAD

**Discussion by bilbothebaggins** [mlange-42/git-graph#66] [Q&A] [answered]
I like git-graph. I like it a lot.

It does one thing I have not seen _any_ git client do, and that is the model feature to control how the branches are laid out via the `persistence` and `order` options and of course the coloring of specific branches.

As a bit of a git noob that has to fight through the visualization of the mess that some of our repos are in, I find it rather fascinating that tools like Fork or Kraken or GitExtensions or even command-line `git log` simply do not seem to allow me basic visualization customization like git-graph does.

So ... I guess I wanted to ask the author whether the feature set of git-graph, or it's tagline "... visualize Git history graphs in a comprehensible way, *following different branching models*" was inspired by something in another tool, or by the lack therof in seemingly "all" other tools.

cheers!

**Comment by mlange-42** [comment, marked as answer]
Thanks, good to read that it helps people.

There is no other tool that I am aware of that does something similar. All the other tools not even respecting *what* was merged into *what* bothered me with their incomprehensible graphs. So this is why `git-graph` exists.

You may, however, want to take a look at [`git-igitt`](https://github.com/mlange-42/git-igitt) if you haven't seen it yet. It is an interactive TUI application based on `git-graph`. It lets you browse the history and view branches, commits and diffs. It is, however, still only for visualization. So you can't manipulate a repository using it.
